hbase-client/src/main/java/org/apache/hadoop/hbase/security/provider/AbstractSaslClientAuthenticationProvider.java

```diff
@@ -48,6 +48,7 @@ public final int hashCode() {
   }
 
   @Override
+  @SuppressWarnings("EqualsUsingHashCode")
```

Comment:
I suggest we do not hide this warning, the implementation seems awkward, maybe it just means comparing the AuthMethod?

Reply:
Sounds good